test/python/common.py

```diff
@@ -243,31 +244,24 @@ def _(*args, **kwargs):
         if SKIP_ONLINE_TESTS:
             raise unittest.SkipTest('Skipping online tests')
 
-        # Try to read the variables from Qconfig.
+        # Cleanup the credentials, as this file is shared by the tests.
+        from qiskit.wrapper import _wrapper
+        _wrapper._DEFAULT_PROVIDER = DefaultQISKitProvider()
+
+        # Attempt to read the standard credentials.
+        discovered_credentials = discover_credentials()
         try:
-            import Qconfig
-            QE_TOKEN = Qconfig.APItoken
-            QE_URL = Qconfig.config['url']
-            QE_HUB = Qconfig.config.get('hub')
-            QE_GROUP = Qconfig.config.get('group')
-            QE_PROJECT = Qconfig.config.get('project')
-        except ImportError:
-            # Try to read them from environment variables (ie. Travis).
-            QE_TOKEN = os.getenv('QE_TOKEN')
-            QE_URL = os.getenv('QE_URL')
-            QE_HUB = os.getenv('QE_HUB')
-            QE_GROUP = os.getenv('QE_GROUP')
-            QE_PROJECT = os.getenv('QE_PROJECT')
-        if not QE_TOKEN or not QE_URL:
-            raise Exception(
-                'Could not locate a valid "Qconfig.py" file nor read the QE '
-                'values from the environment')
-
-        kwargs['QE_TOKEN'] = QE_TOKEN
-        kwargs['QE_URL'] = QE_URL
-        kwargs['hub'] = QE_HUB
-        kwargs['group'] = QE_GROUP
-        kwargs['project'] = QE_PROJECT
+            credentials = next(iter(discovered_credentials.values()))
```

Comment:
Why `next(iter(discovered_credentials.values()))`? Why not `discovered_credentials.values()[0]`?

Reply:
As mentioned in private, due to the fact that `discover_credentials()` can technically return more than one credentials, and more importantly, because it might return no credentials - it felt more pythonic to use that approach. Nevertheless, since the scope of the PR has been reduced and in practice the only relevant account for the tests (and the one likely to exist) is for `IBMQProvider`, I have narrowed the block to account for that.